lib/Sema/ConstraintSystem.cpp

```diff
@@ -2880,6 +2880,58 @@ static bool diagnoseConflictingGenericArguments(ConstraintSystem &cs,
   return diagnosed;
 }
 
+/// Diagnose ambiguity related to overloaded declarations where only
+/// *some* of the overload choices have ephemeral pointer warnings/errors
+/// associated with them. Such situations have be handled specifically
+/// because ephemeral fixes do not affect the score.
+///
+/// If all of the overloads have ephemeral fixes associated with them
+/// it's much easier to diagnose through notes associated with each fix.
+static bool
+diagnoseAmbiguityWithEphemeralPointers(ConstraintSystem &cs,
+                                       ArrayRef<Solution> solutions) {
+  llvm::MapVector<Expr *, ValueDecl *> ambiguities;
+
+  bool allSolutionsHaveFixes = true;
+  for (const auto &solution : solutions) {
+    if (solution.Fixes.empty()) {
+      allSolutionsHaveFixes = false;
+      continue;
+    }
+
+    for (const auto *fix : solution.Fixes) {
+      if (fix->getKind() != FixKind::TreatEphemeralAsNonEphemeral)
+        return false;
+
+      // Ephemeral pointers are only possible in argument positions.
+      if (auto *anchor = simplifyLocatorToAnchor(fix->getLocator())) {
+        anchor = cast<InOutExpr>(anchor)->getSubExpr();
```

Comment:
Good point, for some reason I though that it would only be possible to get an ephemeral pointer if the argument is explicitly `InOutExpr`, I'm going to fix that...

Reply:
I think only possible situation when implicit `inout-to-pointer` is allowed is operator invocations where it's allowed to elide `&` in argument.